packages/xo-server/src/xo-mixins/pool.js

```diff
@@ -0,0 +1,75 @@
+import { difference, flatten, uniq } from 'lodash'
+
+export default class Pools {
+  constructor(xo) {
+    this._xo = xo
+  }
+
+  async mergeInto({ sources: sourceIds, target, force }) {
+    const { _xo } = this
+    const targetHost = _xo.getObject(target.master)
+    const sources = []
+    const sourcePatches = {}
+    for (const sourceId of sourceIds) {
+      const source = _xo.getObject(sourceId)
+      const sourceHost = _xo.getObject(source.master)
+      if (sourceHost.productBrand !== targetHost.productBrand) {
+        throw new Error(
+          `a ${sourceHost.productBrand} pool cannot be merged into a ${
+            targetHost.productBrand
+          } pool`
+        )
+      }
+      sources.push(source)
+      sourcePatches[sourceId] = sourceHost.patches
+    }
+
+    const targetRequiredPatches = uniq(
+      flatten(
+        await Promise.all(
+          sources.map(({ master }) =>
+            _xo.getPatchesDifference(master, target.master)
+          )
+        )
+      )
+    )
+
+    const promises = []
+    if (targetRequiredPatches.length > 0) {
+      const targetXapi = _xo.getXapi(target)
+      promises.push(
+        targetXapi.installPatches({
```

Comment:
For XCP-ng hosts, this will install all the patches, no matter what we pass. Is that an issue? Have you tested it? Should we have a special case for XCP-ng to always install all the patches on every hosts?